protoc-gen-twirp/generator.go

```diff
@@ -966,14 +968,22 @@ func (t *twirp) generateServerJSONMethod(service *descriptor.ServiceDescriptorPr
 	methName := stringutils.CamelCase(method.GetName())
 	t.P(`func (s *`, servStruct, `) serve`, methName, `JSON(ctx `, t.pkgs["context"], `.Context, resp `, t.pkgs["http"], `.ResponseWriter, req *`, t.pkgs["http"], `.Request) {`)
 	t.P(`  var err error`)
+	t.P()
+	t.P(`  defer func() {`)
+	t.P(`    closeErr := req.Body.Close()`)
```

Comment:
The net/http.Server will take care of closing the body. I don't see the benefit of closing it ourselves here, and the extra code and path to calling the Error hook makes these functions more complex.

From `go doc http.Request.Body`,

> The Server will close the request body. The ServeHTTP Handler does not need to.

I'd prefer that you delete this section (and the matching one in `generateServerProtobufMethod`), but it's up to you.

Reply:
Agreed, will do.